git/githistory/log/log.go

```diff
@@ -0,0 +1,197 @@
+package log
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+	"sync"
+
+	"github.com/git-lfs/git-lfs/tools"
+	"github.com/olekukonko/ts"
+)
+
+// Logger logs a series of tasks to an io.Writer, processing each task in order
+// until completion .
+type Logger struct {
+	// sink is the writer to write to.
+	sink io.Writer
+
+	// widthFn is a function that returns the width of the terminal that
+	// this logger is running within.
+	widthFn func() int
+
+	// queue is the incoming, unbuffered queue of tasks to enqueue.
+	queue chan Task
+	// tasks is the set of tasks to process.
+	tasks chan Task
```

Comment:
Thanks for pointing this out. I added the `queue` and `tasks` channel so that new tasks could be enqueued before the current task is finished processing.

Looking back through this, I realized that I removed the callsite in the code that originally necessitated this, but I think if we want to use this package as a general purpose logging utility for LFS, it's worthwhile to keep the extra complexity.

I added a test to ensure that this non-blocking behavior is in place. Before 930c151:

```
~/g/git-lfs (migrate-logger) $ go test -v ./git/githistory/log -run TestLoggerLogsMultipleTasksWithoutBlocking
=== RUN   TestLoggerLogsMultipleTasksWithoutBlocking
# ...
```

and with 930c151:

```
~/g/git-lfs (migrate-logger) $ go test -v ./git/githistory/log -run TestLoggerLogsMultipleTasksWithoutBlocking
=== RUN   TestLoggerLogsMultipleTasksWithoutBlocking
--- PASS: TestLoggerLogsMultipleTasksWithoutBlocking (0.00s)
PASS
ok      github.com/git-lfs/git-lfs/git/githistory/log   0.013s
```